File: game-prototype/src/types/actions/comboAction.ts
import { DiceActions } from './diceAction';
import type { DiceAction } from './diceAction';

export type ComboAction = {
    name: string;
    description: string;
    requiredActions: DiceAction[];
};

type ComboActionType = 'chargedAttack' | 'parry';

export const ComboActions: Record<ComboActionType, ComboAction> = {
    chargedAttack: {
        name: 'Charged Attack',
        description: 'Charge up an attack to deal massive damage.',
        requiredActions: [DiceActions.PhysicalAttack, DiceActions.PhysicalAttack],
    },
    parry: {
        name: 'Parry',
        description: 'Block and counter an incoming attack.',
        requiredActions: [DiceActions.Defend, DiceActions.PhysicalAttack],
    }
} as const;

File: game/models/actions/ComboAction.cs
using Godot;
using System.Collections.Generic;

[Tool]
public partial class ComboAction : Resource {
	[Export]
	public string Name { get; set; }
	[Export]
	public string Description { get; set; }
	[Export]
	public Godot.Collections.Array<DiceAction> RequiredActions { get; set; } = new Godot.Collections.Array<DiceAction>();

	public ComboAction() { }

	public ComboAction(string name, string description, Godot.Collections.Array<DiceAction> requiredActions) {
		Name = name;
		Description = description;
		RequiredActions = requiredActions;
	}
}